File: cpp/CowGymnastics.cpp
#include<iostream>
using namespace std;

int k, n;
int datas[10][20];

bool better(int a, int b, int session)
{
  int apos, bpos;
  for (int i = 0; i < n; i++) {
    if (datas[session][i] == a) apos = i;
    if (datas[session][i] == b) bpos = i;
  }
  return apos < bpos;
}
 
int Nbetter(int a, int b)
{
  int total = 0;
  for (int session=0; session< k; session++)
    if (better(a,b,session)) total++;
  return total;
}

int main(){
    freopen("gymnastics.in", "r", stdin);
    freopen("gymnastics.out", "w", stdout);

    cin >> k >> n;
    for(int i = 0; i < k; i++){
        for(int j = 0; j < n; j++){
            cin >> datas[i][j];
        }
    }

    int answer = 0;
    for (int a=1; a<=n; a++)
        for (int b=1; b<=n; b++)
            if (Nbetter(a,b) == k) 
                answer++;
    
    cout << answer << endl;
}

File: java/CowGymnastics.java
import java.io.*;
import java.util.*;

class CowGymnastics {

    public static int numP, numC;
    public static int[][] rankings;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new FileReader("gymnastics.in"));
        PrintWriter pw = new PrintWriter(new FileWriter("gymnastics.out"));
        StringTokenizer st = new StringTokenizer(br.readLine());

        numP = Integer.parseInt(st.nextToken());
        numC = Integer.parseInt(st.nextToken());
        rankings = new int[10][20];
        
        for(int i = 0; i < numP; i++){
            st = new StringTokenizer(br.readLine());
            for(int j = 0; i < numC; i++){
                rankings[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        int ctr = 0;
        for(int x = 1; x <= numC; x++){
            for(int y = 1; y <= numC; y++){
                if(n_better(x, y) == numP){
                    ctr++;
                }
            }
        }

        pw.println(ctr);
        pw.close();
    }

    public static int n_better(int a, int b){
        int total = 0;
        for(int i = 0; i < numP; i++){
            if(better(a, b, i)){
                total++;
            }
        }
        return total;
    }

    public static boolean better(int a, int b, int i){
        int aPos = 0; 
        int bPos = 0;
        for(int j = 0; j < numC; j++){
            if(rankings[i][j] == a){
                aPos = j;
            }
            if(rankings[i][j] == b){
                bPos = i;
            }
        }
        return aPos < bPos;
    }
}